### PR TITLE
Use tinyvec for actor id

### PR DESCRIPTION
**Pull request by jeffa5**
Most of the time users will not be using their own identifiers and so
use random uuids. These can fit nicely on the stack for a speedup.
TinyVec allows us to capture this mostly stack, sometimes heap
behaviour.

### CODE REVIEW

File: automerge-protocol/Cargo.toml
Comment (alexjg) [inline automerge-protocol/Cargo.toml:17]: Have you checked that this builds okay in a `no_std` environment? It looks like it should but I'm not certain.

Comment (jeffa5) [reply automerge-protocol/Cargo.toml:17]: I don't think automerge builds in a `no_std` environment for a while (at least running `cargo nono check --package automerge` gives lots of issues). Is there a concrete use case for use in `no_std` at the moment? I can tell it might be nice but at this point I think it will be quite a lot of work to get there.

**Comment by orionz** [comment]
Have you measured the impact on performance?

**Comment by jeffa5** [comment]
@orionz I'm seeing about a 7-11% improvement on the B1 benchmarks compared to main

File: automerge-protocol/Cargo.toml
Comment (alexjg) [reply automerge-protocol/Cargo.toml:18]: Fair. Let's not worry about that now then.

**Comment by alexjg** [comment]
I'm happy to merge this once conflicts are fixed. I'll wait until the conflicts are fixed to mark it as approved though or my approval will get dismissed by the updates.

**Comment by jeffa5** [comment]
I'll wait on https://github.com/Lokathor/tinyvec/pull/146 to work on this as we should still retain the arbitrary functionality.

**Comment by jeffa5** [comment]
My TinyVec PR got merged so I've updated this PR to use the git version.

**Review by alexjg — APPROVED** [review]

**Pull request merged**
